python/src/phylanx.cpp

```diff
@@ -497,36 +535,390 @@ PYBIND11_MODULE(_phylanx, m)
+                for (int j = 0; j < ny; j++)
+                {
+                    m(i, j) = m0 + dx * i + dy * j;
+                }
+            }
+            return hpx::threads::run_as_hpx_thread([&]() {
+                using namespace phylanx::execution_tree;
+                return primitive{hpx::local_new<primitives::variable>(
+                    phylanx::ir::node_data<double>{m})};
+            });
+        },
+        "create a new variable from a new matrix from linear coefficients m0, "
+        "dx, and dy");
+
+    execution_tree.def("file_read_csv",
+        [](std::string fname) {
+            return hpx::threads::run_as_hpx_thread([&]() {
+                using namespace phylanx::execution_tree;
+                auto frc =
+                    phylanx::execution_tree::primitives::file_read_csv{{fname}};
+                std::vector<primitive_argument_type> args;
+                primitive_argument_type result = frc.eval(args).get();
+                return primitive{
+                    hpx::local_new<primitives::variable>(result.variant())};
+            });
+        },
+        "Read in the contents of a csv file");
+
+    execution_tree.def("slice",
+        [](phylanx::execution_tree::primitive const& obj, std::int64_t x0,
+            std::int64_t xN, std::int64_t y0, std::int64_t yN) {
+            return hpx::threads::run_as_hpx_thread([&]() {
+                using namespace phylanx::execution_tree;
+                auto frc =
+                    phylanx::execution_tree::primitives::slicing_operation{
+                        {obj, x0, xN, y0, yN}};
+                std::vector<primitive_argument_type> args;
+                primitive_argument_type result = frc.eval(args).get();
+                return primitive{
+                    hpx::local_new<primitives::variable>(result.variant())};
+            });
+        },
+        "Take a slice of a 2d array");
+
+    execution_tree.def("zeros",
+        [](std::size_t nx) {
+            blaze::DynamicVector<double> v;
+            v.resize(nx);
+            return hpx::threads::run_as_hpx_thread([&]() {
+                using namespace phylanx::execution_tree;
+                return primitive{hpx::local_new<primitives::variable>(
+                    phylanx::ir::node_data<double>{v})};
+            });
+        },
+        "create a new variable from a vector of zeros");
+
+    execution_tree.def("zeros",
+        [](std::size_t nx, std::size_t ny) {
+            blaze::DynamicMatrix<double> m{nx, ny};
+            m = 0.0;
+            return hpx::threads::run_as_hpx_thread([&]() {
+                using namespace phylanx::execution_tree;
+                return primitive{hpx::local_new<primitives::variable>(
+                    phylanx::ir::node_data<double>{m})};
+            });
+        },
+        "create a new variable from a matrix of zeros");
+
+    execution_tree.def(
+        "phylisp_eval", expression_compiler<>, expression_compiler_help);
```

Comment:
We shouldn't call the language `phylisp`. Why I see where this comes from, I believe it's not a 'lisp-ish' language.

Reply:
Do you have a suggested name?

Reply:
Just `eval` should be fine in this context. 

Wrt the name for our little language - I heard somebody calling it `PhySL`, but I'm really impartial and would like to hear other oppinions.